// cloudflare@4.2.0/resources/zero-trust/devices/networks.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeviceNetworksSinglePage = exports.Networks = void 0;
const resource_1 = require("../../../resource.js");
const pagination_1 = require("../../../pagination.js");
class Networks extends resource_1.APIResource {
    /**
     * Creates a new device managed network.
     */
    create(params, options) {
        const { account_id, ...body } = params;
        return this._client.post(`/accounts/${account_id}/devices/networks`, { body, ...options })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Updates a configured device managed network.
     */
    update(networkId, params, options) {
        const { account_id, ...body } = params;
        return this._client.put(`/accounts/${account_id}/devices/networks/${networkId}`, {
            body,
            ...options,
        })._thenUnwrap((obj) => obj.result);
    }
    /**
     * Fetches a list of managed networks for an account.
     */
    list(params, options) {
        const { account_id } = params;
        return this._client.getAPIList(`/accounts/${account_id}/devices/networks`, DeviceNetworksSinglePage, options);
    }
    /**
     * Deletes a device managed network and fetches a list of the remaining device
     * managed networks for an account.
     */
    delete(networkId, params, options) {
        const { account_id } = params;
        return this._client.getAPIList(`/accounts/${account_id}/devices/networks/${networkId}`, DeviceNetworksSinglePage, { method: 'delete', ...options });
    }
    /**
     * Fetches details for a single managed network.
     */
    get(networkId, params, options) {
        const { account_id } = params;
        return this._client.get(`/accounts/${account_id}/devices/networks/${networkId}`, options)._thenUnwrap((obj) => obj.result);
    }
}
exports.Networks = Networks;
class DeviceNetworksSinglePage extends pagination_1.SinglePage {
}
exports.DeviceNetworksSinglePage = DeviceNetworksSinglePage;
Networks.DeviceNetworksSinglePage = DeviceNetworksSinglePage;
